package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1519,6 +1519,8 @@ def center_of_mass(group, wrap=False, unwrap=False, compound='group'):
            is deprecated and will be removed in version 3.0.
         """
         atoms = group.atoms
+        if not len(atoms):
```

Comment:
This can be simplified to
```python
if not atoms:
```
because the truthiness of an empty AtomGroup is `False`.

Reply:
Fair enough, the weird error come into play with the shape_parameter and asphericity methods. The center_of_mass doesn't have an issue. 
This could also be resolved with an if statement inside shape_parameter and asphericity with: 
```
if not group.atoms:
    return np.array([])
```

Reply:
One approach to make this contribution much more powerful would be to add a simple decorator which raises an error if the atomgroup is empty.

Reply:
This is a great idea something like `@check_atomgroup_not_empty`!